eva/udfs/yolo_object_detector.py

```diff
@@ -157,26 +159,29 @@ def forward(self, frames: Tensor) -> pd.DataFrame:
         """
         # Stacking all frames, and changing to numpy
         # because of yolov5 error with Tensors
-
         outcome = []
         # Convert to HWC
         # https://github.com/ultralytics/yolov5/blob/3e55763d45f9c5f8217e4dad5ba1e6c1f42e3bf8/models/common.py#L658
         frames = torch.permute(frames, (0, 2, 3, 1))
-        predictions = self.model([its.cpu().detach().numpy() * 255 for its in frames])
-
-        for i in range(frames.shape[0]):
-            single_result = predictions.pandas().xyxy[i]
-            pred_class = single_result["name"].tolist()
-            pred_score = single_result["confidence"].tolist()
-            pred_boxes = single_result[["xmin", "ymin", "xmax", "ymax"]].apply(
-                lambda x: list(x), axis=1
-            )
+        list_of_numpy_images = [its.cpu().detach().numpy() * 255 for its in frames]
+        predictions = self.predict_func(list_of_numpy_images)
+
+        for pred in predictions:
+            single_result = pred.boxes
+            pred_class = [
+                self.predict_func.names[i] for i in single_result.cls.tolist()
+            ]
+            pred_score = single_result.conf.tolist()
+            pred_boxes = single_result.xyxy.tolist()
+
+            sorted_list = list(map(lambda i: i < self.threshold, pred_score))
```

Comment:
get rid of the threshold from here and let us use confidence to control it in queries. Btw, is there a default value?

Reply:
If there is no default value, then we can stick with 0.3.

Reply:
there is no default value for the model